lib/active_model/serializer.rb

```diff
@@ -108,20 +110,32 @@ def self.digest_caller_file(caller_line)
       Digest::MD5.hexdigest(serializer_file_contents)
     end
 
-    def self.get_serializer_for(klass)
+    def self.get_serializer_for(klass, parent_serializer, controller)
       serializers_cache.fetch_or_store(klass) do
         serializer_class_name = "#{klass.name}Serializer"
-        serializer_class = serializer_class_name.safe_constantize
+
+        nested_serializer_class_name = "#{self}::#{serializer_class_name}"
+        serializer_class = nested_serializer_class_name.safe_constantize
+
+        if parent_serializer
+          namespaced_serializer_class_name = "#{parent_serializer.root_serializer.class.name.deconstantize}::#{serializer_class_name}"
+          serializer_class ||= namespaced_serializer_class_name.safe_constantize
+        elsif controller
+          controller_namespaced_serializer_class_name = "#{controller.class.name.deconstantize}::#{serializer_class_name}"
+          serializer_class ||= controller_namespaced_serializer_class_name.safe_constantize
```

Comment:
what if this serializer isn't found? like, what if someone wants to use a straight up `PostSerializer` in `/blog/id/posts`?

tests for this would be great :-)